prov/rxm/src/rxm_ep.c

```diff
@@ -475,8 +478,10 @@ static int rxm_join_coll(struct fid_ep *ep, const void *addr, uint64_t flags,
 	ofi_spin_lock(&rxm_mc->state_lock);
 	rxm_mc->state = RXM_MC_UTIL_STARTED;
 	ofi_spin_unlock(&rxm_mc->state_lock);
+	peer_context.mc_fid = &rxm_mc->mc_fid;
 	ret = fi_join_collective(rxm_ep->util_coll_ep, c_addr->coll_addr,
-			c_addr->set, flags, &rxm_mc->util_coll_mc_fid, rxm_mc);
+		c_addr->set, flags | FI_PEER, &rxm_mc->util_coll_mc_fid,
+		&peer_context);
```

Comment:
We can't pass the address of a stack variable as the context to an async call.

Reply:
This has been modeled on rxm_cq_open().
The FI_PEER flag informs us to handle the context in a different way (not as persistent one)